src/lib/update-alternatives/UI/alternatives_dialog.rb

```diff
@@ -47,25 +56,33 @@ def set_handler
       finish_dialog()
     end
 
+    def auto_handler
+      log.info("You have use the \"Set automatic mode\" button")
+      finish_dialog()
+    end
+
+    def alternatives_handler
+      selected_alternative = Yast::UI.QueryWidget(Id(:alternatives), :CurrentItem)
+      Yast::UI.ChangeWidget(Id(:slaves), :Value, @mock_slaves[selected_alternative])
+    end
+
     def create_alternatives_table
       Table(
-        Id(:alternatives_table),
+        Id(:alternatives),
+        Opt(:notify, :immediate),
         Header(_("Alternative"), _("Priority")),
         [
-          Item(Id(:java18), "/usr/lib64/jvm/jre-1.8.0-openjdk/bin/java", "1805"),
-          Item(Id(:java17), "/usr/lib64/jvm/jre-1.7.0-openjdk/bin/java", "1705")
+          Item(Id(:vim), "/usr/bin/vim.basic", "50"),
+          Item(Id(:ed), "/bin/ed", "-100")
         ]
       )
     end
 
-    def create_slaves_table
-      Table(
-      Id(:alternatives_table),
-      Header(_("Slave name"), _("Slave alternave")),
-      [
-        Item(Id(:slave1), "slave1", "path/to/slave1/alternative/for/1.8"),
-        Item(Id(:slave2), "slave2", "path/to/slave2/alternative/for/1.8")
-      ]
+    def footer
+      HBox(
+        PushButton(Id(:set), _("Set alternative")),
+        PushButton(Id(:auto), _("Set automatic mode")),
+        PushButton(Id(:cancel), _("Cancel"))
```

Comment:
to save translators work, It is recommended for common labels to use Label module -  https://github.com/yast/yast-yast2/blob/8eb9214265fdf8afcae1531c5a0309799028d79d/library/general/src/modules/Label.rb#L50

also notice that there is used `&` which specify which character is used as shortcut. It is good to use it, as it helps with having stable keyboard shortcuts.